packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -218,24 +218,28 @@ class FlutterDevice {
     return completer.future;
   }
 
+  // TODO(jonahwilliams): remove once all callsites are updated.
+  VMService get oldVmService => vmService as VMService;
```

Comment:
I found myself slightly blocked on testing:

I can't move everything to the new vm_service yet, but updating the tests was becoming increasingly difficult due to the increased requirements for mocking.

I split the vm_service usage between the new vm service and the old vm service. The functionality that depends on the new vm service API can be tested using the vm service host and skipping the mocks,  while the old functionality can use the existing mocks.

As more functionality is migrated to use the new vm service, I can replace the mocks and eventually delete them

Reply:
"old" and "new" are not good names, don't assume your "new" won't become "old" aka "reallyOld" "legacy", I'm sure you've seen examples of this live for years and years. 

Also, make this comment more helpful, what callsites, what are you replacing?  I really wish we had https://github.com/flutter/flutter/issues/37519, it shouldn't really be a note-to-self.  Check out how many TODOs we have in the repo "assigned" to people who have left the team.

Reply:
I mean how many TODOS could I have in my name ... (5 pages 🤦 )

I updated the name to be more descriptive, tell me if you like it. Though hopefully at the rate I'm going it will be gone soon

Reply:
Sounds like if we do get #37519, it will be as a new lint, i.e. `new_flutter_style_todos`